dask/config.py

```diff
@@ -586,7 +588,48 @@ def check_deprecations(key: str, deprecations: dict = deprecations):
         return key
 
 
+def serialize(data):
+    """Serialize config data into a string.
+
+    Typically used to pass config via the ``DASK_INTERNAL_INHERIT_CONFIG`` environment variable.
+
+    Parameters
+    ----------
+    data: json-serializable object
+        The data to serialize
+
+    Returns
+    -------
+    serialized_data: str
+        The serialized data as a string
+
+    """
+    return base64.urlsafe_b64encode(json.dumps(data).encode()).decode()
+
+
+def deserialize(data):
+    """De-serialize config data into the original object.
+
+    Typically when receiving config via the ``DASK_INTERNAL_INHERIT_CONFIG`` environment variable.
+
+    Parameters
+    ----------
+    data: str
+        String serialied by :func:`dask.config.serialize`
+
+    Returns
+    -------
+    deserialized_data: obj
+        The de-serialized data
+
+    """
+    return json.loads(base64.urlsafe_b64decode(data.encode()).decode())
+
+
 def _initialize():
+    if "DASK_INTERNAL_INHERIT_CONFIG" in os.environ:
```

Comment:
Does this take precedence over config in files, as other environment variables do?

I would suggest the block belongs better in function `collect`, which is where other environment variables are considered, and allows you to pass in a custom env dictionary.

Reply:
Thanks @martindurant that makes a lot of sense. Updated!